### PR TITLE
EC Adjacent Vertices Calculator

### PR DESCRIPTION
**Pull request by ericthewry**
Hey @sf-interns/owners, here's a crack at the Service for adjacent vertices, let me know if you see anything that needs changing.

Make sure to look at the tests. They might give a jumping-off point for how to structure the tests for the future.

### CODE REVIEW

File: app/services/AdjacentCellsCalculator.coffee [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+class AdjacentCellsCalculator
+
+  # Param grid is a GameGrid
+  constructor: (@grid, @cells, @x, @y) ->
+
+  # Gets the adjacent cells
+  calculate: () => 
+    for i in [@y - 1, @y, @y + 1]
+      for j in [@x - 1, @x, @x + 1]
+        continue if i is @y and j is @x
```
Comment (shubhamgogna) [inline app/services/AdjacentCellsCalculator.coffee:12]: `unless i is @y and j is @x`

Comment (ericthewry) [inline app/services/AdjacentCellsCalculator.coffee]: I think `continue` skips to the next iteration of the loop. So here I'm trying to skip excecution for the current cell. Unless `continue` does something else?

<div>-------- Original message --------</div><div>From: Shubham Gogna <notifications@github.com> </div><div>Date:07/20/2015  9:07 AM  (GMT-08:00) </div><div>To: sf-interns/mathswipe <mathswipe@noreply.github.com> </div><div>Cc: Eric Campbell <eric.campbell@originate.com> </div><div>Subject: Re: [mathswipe] EC Adjacent Vertices Calculator (#8) </div><div>
</div>In app/services/AdjacentCellsCalculator.coffee:

> @@ -0,0 +1,24 @@
> +Tuple = require "../models/Tuple"
> +
> +class AdjacentCellsCalculator
> +
> +  # Param grid is a GameGrid
> +  constructor: (@grid, @cells, @x, @y) ->
> +
> +  # Gets the adjacent cells
> +  calculate: () => 
> +    for i in [@y - 1, @y, @y + 1]
> +      for j in [@x - 1, @x, @x + 1]
> +        continue if i is @y and j is @x
unless i is @y and j is @x

File: app/models/Tuple.coffee [hunk 2]
```diff
@@ -0,0 +1,14 @@
+class Tuple
+
+  constructor: (@x,@y) ->
+
+  equals: (otherTuple) =>
+    return false if otherTuple is null
+    @x is otherTuple.x and @y is otherTuple.y
+
+  isElementOf: (tuples) =>
```
Comment (ericthewry) [inline app/models/Tuple.coffee]: `isElementOf` should be in `TupleSet` as `contains`

**Comment by shubhamgogna** [comment]
LGTM :shipit:

**Pull request merged**